db/migrate/20210621234939_backfill_email_log_topic_id.rb

```diff
@@ -0,0 +1,17 @@
+# frozen_string_literal: true
+
+class BackfillEmailLogTopicId < ActiveRecord::Migration[6.1]
+  def up
+    DB.exec(<<~SQL)
+      UPDATE email_logs AS el
```

Comment:
I'm alittle uneasy about updating so many columns at once since we may end up locking the table and temporary cause email sending to fail. On mailing list enabled sites, the email logs table is huge. Are we able to update in batches instead?

Reply:
I am going to test this query on a couple of sites first (with ROLLBACK) to see the impact...I have not been able to yet because the deploy keeps failing

Reply:
I will look into the locking...would really prefer not to overcomplicate this query, I have done big UPDATE FROM queries in the past without issue.

Reply:
@martin-brennan Btw I think I remember why I dropped the `topic_id` column previously. We can basically get the `topic_id` via the `post` association. Is there another need for the `topic_id` here?

Reply:
@tgxworld relying on an association is kind of a pain because you have to join across tables just to get the topic ID. Would rather just have the topic ID on this table like we do for `IncomingEmail`, it will make things a lot easier. Filtering all outbound emails for a topic becomes a lot easier.

I still haven't gotten around to testing out this migration, doing that today and will batch if needed.

Reply:
@tgxworld Moved this update into batches and I tested on Meta with a rollback; this is what I did there:


```ruby
Benchmark.measure do
  ActiveRecord::Base.transaction do
    BATCH_SIZE = 30000
    offset = 0
    email_log_count = EmailLog.count

    loop do
      DB.exec(<<~SQL, offset: offset, batch_size: BATCH_SIZE)
          WITH cte AS (
            SELECT topic_id, post_id
            FROM email_logs
            ORDER BY id
            LIMIT :batch_size
            OFFSET :offset
          )
          UPDATE email_logs
          SET topic_id = posts.topic_id
          FROM cte
          INNER JOIN posts ON posts.id = cte.post_id
          WHERE email_logs.post_id = cte.post_id
      SQL

      offset += BATCH_SIZE
      break if offset > (email_log_count + BATCH_SIZE * 2)
    end
    raise ActiveRecord::Rollback
  end
end
```

The results were that it took 34s to run this on Meta. Meta has `506230` email log records right now, so that's about 2s per batch of 30,000. I am actually a little wary of removing the `topic` method in this PR, it leaves us with no wiggle room if this migration goes bad, I think I will remove that in another PR.

What are your thoughts on this now?